packages/experiments-realm/fields/amount-with-currency.gts

```diff
@@ -0,0 +1,115 @@
+import NumberField from 'https://cardstack.com/base/number';
+import { FieldDef, field, contains } from 'https://cardstack.com/base/card-api';
+import { Component } from 'https://cardstack.com/base/card-api';
+import { CurrencyField } from './currency';
+import { action } from '@ember/object';
+import { BoxelInputGroup } from '@cardstack/boxel-ui/components';
+import { guidFor } from '@ember/object/internals';
+
+class View extends Component<typeof AmountWithCurrency> {
+  get formatNumberWithSeparator() {
+    const num = this.args.model.amount;
+    const currencySymbol = this.args.model.currency?.symbol;
+
+    if (num === null || num === undefined) {
+      return '';
+    }
+
+    return `${currencySymbol} ${num.toLocaleString('en-US')}`;
+  }
+
+  <template>
+    {{this.formatNumberWithSeparator}}
+  </template>
+}
+
+class Edit extends Component<typeof AmountWithCurrency> {
+  get id() {
+    return guidFor(this);
+  }
+
+  @action
+  setAmount(val: number) {
+    let newModel = new AmountWithCurrency();
+    newModel.amount = val;
+    newModel.currency.code = newModel.currency.code || 'USD';
+    this.args.set(newModel);
+  }
+
+  @action
+  setCurrency(val: CurrencyField) {
+    let newModel = new AmountWithCurrency();
+    newModel.amount = this.args.model.amount as number;
```

Comment:
@tintinthong lint is complaining to have a type on `amount`, so I keep the changes here